### ISSUE TITLE
Can't update YAML keeping newlines

### ISSUE THREAD

**Issue by suzuki-shunsuke** [goccy/go-yaml#285]
Thank you for your great project and sorry for bothering you.
I have a question about the usage of this library.

I would like to update a YAML file keeping comments, indents, and new lines.
It succeeded to update a YAML file keeping comments and indents, but new lines were removed.

go.mod

<details>

```
module foo

go 1.17

require github.com/goccy/go-yaml v1.9.5

require (
	github.com/fatih/color v1.10.0 // indirect
	github.com/mattn/go-colorable v0.1.8 // indirect
	github.com/mattn/go-isatty v0.0.12 // indirect
	golang.org/x/sys v0.0.0-20200223170610-d5e6a3e2c0ae // indirect
	golang.org/x/xerrors v0.0.0-20200804184101-5ec99f83aff1 // indirect
)
```

</details>

main.go

https://go.dev/play/p/SsSu3_TyYeK

I found `*ast.File.String()` doesn't keep new lines.

https://go.dev/play/p/dT9pQ52wTrD

Expected Result

```yaml
---
# aqua - Declarative CLI Version Manager
# https://aquaproj.github.io/
registries:
  - type: standard
    ref: v1.3.0 # renovate: depName=aquaproj/aqua-registry

packages:
  - name: suzuki-shunsuke/tfcmt@v3.0.1
```

Actual Result

```yaml
---
# aqua - Declarative CLI Version Manager
# https://aquaproj.github.io/
registries:
  - type: standard
    ref: v1.3.0 # renovate: depName=aquaproj/aqua-registry
packages:
  - name: suzuki-shunsuke/tfcmt@v3.0.1
```

I'll really appreciate it if you have any advice.
Thank you.

**Comment by suzuki-shunsuke** [comment]
I found [printer.PrintTokens](https://pkg.go.dev/github.com/goccy/go-yaml@v1.9.5/printer#Printer.PrintTokens) keeps new lines, so I'm looking for how to update token.Tokens.

**Comment by zellyn** [comment]
@suzuki-shunsuke would you post here if you figure it out? Thanks!

**Comment by suzuki-shunsuke** [comment]
Unfortunately, I don't figure it out yet.

Related issue: https://github.com/aquaproj/aqua/issues/584

**Comment by marco-m-pix4d** [comment]
Adding the one word "newline" to make it easier to search.

**Comment by gabe565** [comment]
It looks like the difference in behavior is because `ast.Node.String` receivers use `Token.Value` to format their values, while `printer.PrintTokens` uses `Token.Origin`. `Token.Value` seems to have the newlines stripped out.  
I don't know enough about the internal functionality to say if it's a good idea to change the `ast.Node.String` receivers to use `Token.Origin`.

For example, if we looks at the internal structure of the following yaml:

```yaml
key1: ""

key2: ""
```

the second key token has the following values:

<img width="427" alt="image" src="https://user-images.githubusercontent.com/7717888/178643410-a7f8034b-484e-43ff-89b2-9127b9a7085f.png">

Sequences behave similarly, with the newline being kept in `Token.Origin` of the `SequenceEntryType`.  
I'm sure it would be possible to build your own visitor that gets passed to `ast.Walk` to keep the newlines in origin, but that would likely get complex quickly.

For someone more familiar with the inner workings of the library: would it be a good idea to stringify mapping keys and sequence entries using `Origin` instead of `Value`?

**Comment by goccy** [comment]
@suzuki-shunsuke 
Thank you for your reporting !! Also, Sorry for the late reply.
This problem has already been fixed with latest release ! Please confirm it !

**Comment by suzuki-shunsuke** [comment]
@goccy 

Thank you for your update!
I've confirmed this issue was solved at 1.13.3, though it wasn't solved at 1.13.2.

I guess this issue was solved by #412 .

- https://github.com/goccy/go-yaml/pull/412
